File: 0020-valid-parentheses/0020-valid-parentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        stack = []
        paras = {'(' : ')', '{' : '}', '[' : ']'}

        for para in s:
            if para in paras:
                stack.append(paras[para])
            elif not stack or para != stack.pop():
                return False

        return not stack

File: 0020-valid-parentheses/0020-valid-parentheses.js
/**
 * @param {string} s
 * @return {boolean}
 */
var isValid = function(s) {
    const openToCloseMap = new Map([
        ['(', ')'],
        ['{', '}'],
        ['[', ']']
    ]);
    
    var stack = [];
    
    for (let para of s){
        if (openToCloseMap.has(para)){
            stack.push(openToCloseMap.get(para));
        } else if((stack.length == 0) || (stack.pop() != para)){
            return false;
        }
    }
    
    return stack.length == 0;
    
};